test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common.Config
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class EdgeConfigBuilder
+    {
+        readonly string deviceId;
+        readonly Dictionary<string, IModuleConfigBuilder> moduleBuilders;
+        Option<(string address, string username, string password)> registry;
+
+        public EdgeConfigBuilder(string deviceId)
+        {
+            this.deviceId = deviceId;
+            this.moduleBuilders = new Dictionary<string, IModuleConfigBuilder>();
+            this.registry = Option.None<(string, string, string)>();
+        }
+
+        public void AddRegistryCredentials(string address, string username, string password)
+        {
+            this.registry = Option.Some((address, username, password));
+        }
+
+        public IModuleConfigBuilder AddEdgeAgent(string image = null)
+        {
+            Option<string> imageOption = Option.Maybe(image);
```

Comment:
nit: should it be better to use Option<string> for image argument?

Reply:
Unfortunately default arguments have to be a compile-time constants, and `Option.None<string>()` is not a compile-time constant. 😢 It was important that the caller be able to write `builder.AddEdgeAgent()` instead of `builder.AddEdgeAgent(Option.None<string>())`, so I had to go with just `string`.